src/components/Reports/GoalCalculator/Shared/GoalCalculatorLayout.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { Divider, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { navBarHeight } from 'src/components/Layouts/Primary/Primary';
+import { multiPageHeaderHeight } from 'src/components/Shared/MultiPageLayout/MultiPageHeader';
+import theme from 'src/theme';
+import { GoalCalculatorStepEnum } from '../GoalCalculatorHelper';
+import { useGoalCalculator } from './GoalCalculatorContext';
+
+const iconPanelWidth = theme.spacing(5);
+
+const MainContent = styled('div')(({ theme }) => ({
+  paddingBlock: theme.spacing(4),
+  width: '100%',
+  height: `calc(100vh - ${navBarHeight} - ${multiPageHeaderHeight})`,
+  overflow: 'scroll',
+}));
+
+const StepTitle = styled(Typography)(({ theme }) => ({
+  marginBottom: 0,
+  marginTop: theme.spacing(1),
+  paddingLeft: theme.spacing(2),
+}));
+
+const StyledDrawer = styled('nav', {
+  shouldForwardProp: (prop) => prop !== 'open',
+})<{ open: boolean }>(({ theme, open }) => ({
+  width: open ? 240 : 0,
+  transition: theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  }),
+  overflow: 'hidden',
+  borderRight: open ? `1px solid ${theme.palette.cruGrayLight.main}` : 'none',
+  [theme.breakpoints.down('sm')]: {
+    position: 'absolute',
+    top: multiPageHeaderHeight,
+    left: `calc(${iconPanelWidth} + 1px)`,
+    height: '100%',
+    backgroundColor: theme.palette.common.white,
+    zIndex: 270,
+  },
+}));
+
+interface GoalCalculatorLayoutProps {
+  sectionListPanel: React.ReactNode;
+  mainContent: React.ReactNode;
+}
+
+/**
+ * This is the layout shared by all goal calculator pages. It renders the icon drawer for changing,
+ * steps, it has a slot for the section list (which is the categories list on most pages), and it
+ * has a slot for the main content.
+ */
+export const GoalCalculatorLayout: React.FC<GoalCalculatorLayoutProps> = ({
+  sectionListPanel,
+  mainContent,
+}) => {
+  const { t } = useTranslation();
+  const iconPanelWidth = theme.spacing(5);
+  const {
+    steps,
+    currentStep,
+    handleStepChange,
+    isDrawerOpen,
+    setDrawerOpen,
+    toggleDrawer,
+    selectedStepId,
+  } = useGoalCalculator();
+
```

Comment:
[Mentoring] Excellent context integration pattern. The destructuring of `useGoalCalculator()` hook and conditional drawer logic demonstrates proper React state management. This pattern of combining context state with local UI state (drawer) is a good example for other developers.